### ISSUE TITLE
ExportToSpreadsheet error when filtering objects in IDSecondary

### ISSUE THREAD

**Issue by bethac07** [CellProfiler/CellProfiler#3379]
In a pipeline which has an IDSecondary module in it where you're filtering primary objects based on the secondary objects (aka when `Discard secondary objects touching the border of the image?` and `Discard the associated primary objects?` are both set to "Yes"), trying to only export specific measurements in ExportToSpreadsheet using the `Press to select measurements` option box gives the following error:

```
Traceback (most recent call last):
  File "CellProfiler\cellprofiler\gui\moduleview.py", line 878, in on_press
  File "CellProfiler\cellprofiler\setting.py", line 1977, in populate_choices
  File "CellProfiler\cellprofiler\setting.py", line 1974, in valid_mc
TypeError: argument of type 'ObjectNameSubscriber' is not iterable
``` 

If you export all measurements it works fine, and the `Press to select measurements` option works fine if `Discard the associated primary objects?` is set to "No", so it's only when the two combine you end up with problems.

Pipeline attached (and images, though those shouldn't be necessary to validate). 

[IDSecondary_ExportToSpreadsheet.zip](https://github.com/CellProfiler/CellProfiler/files/1459092/IDSecondary_ExportToSpreadsheet.zip)

**Comment by bethac07** [comment]
Appears to be fixed in 3.1.9, probably by #3685

**Comment by tischi** [comment]
Yes, in 3.1.9 we do not get this error.
